convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,219 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+import shutil
+
+from convert2rhel import actions, grub, systeminfo
+from convert2rhel.grub import CENTOS_EFIDIR_CANONICAL_PATH, RHEL_EFIDIR_CANONICAL_PATH
+
+
+logger = logging.getLogger(__name__)
+
+
+class NewDefaultEfiBin(actions.Action):
+    id = "NEW_DEFAULT_EFI_BIN"
+
+    def run(self):
+        """Check that the expected RHEL UEFI binaries exist."""
+        super(NewDefaultEfiBin, self).run()
+
+        if not grub.is_efi():
+            logger.info("BIOS detected. Nothing to do.")
+            return
+
+        new_default_efibin = None
+        for filename in grub.DEFAULT_INSTALLED_EFIBIN_FILENAMES:
+            efi_path = os.path.join(RHEL_EFIDIR_CANONICAL_PATH, filename)
+            if os.path.exists(efi_path):
+                logger.info("UEFI binary found: %s" % efi_path)
+                new_default_efibin = efi_path
+                break
+            logger.debug("UEFI binary %s not found. Checking next possibility..." % efi_path)
+        if not new_default_efibin:
+            self.set_result(
+                level="ERROR",
+                id="RHEL_UEFI_BINARIES_DO_NOT_EXIST",
+                title="RHEL UEFI binaries do not exist",
+                description="None of the expected RHEL UEFI binaries exist.",
+                diagnosis="The migration of the bootloader setup was not successful.",
+                remediations=(
+                    "Do not reboot your machine before doing a manual check of the\n"
+                    "bootloader configuration. Ensure that grubenv and grub.cfg files\n"
+                    "are present in the %s directory and that\n"
+                    "a new bootloader entry for Red Hat Enterprise Linux exists\n"
+                    "(check `efibootmgr -v` output).\n"
+                    "The entry should point to '\\EFI\\redhat\\shimx64.efi'." % grub.RHEL_EFIDIR_CANONICAL_PATH
+                ),
+            )
+
+
+class EfibootmgrUtilityInstalled(actions.Action):
+    id = "EFIBOOTMGR_UTILITY_INSTALLED"
+    dependencies = ("NEW_DEFAULT_EFI_BIN",)
+
+    def run(self):
+        """Check if the Efibootmgr utility is installed"""
+        super(EfibootmgrUtilityInstalled, self).run()
+
+        if not os.path.exists("/usr/sbin/efibootmgr"):
+            self.set_result(
+                level="ERROR",
+                id="EFIBOOTMGR_UTILITY_NOT_INSTALLED",
+                title="Efibootmgr utility is not installed",
+                description="The /usr/sbin/efibootmgr utility is not installed.",
+                remediations="Install the efibootmgr utility via YUM/DNF.",
```

Comment:
Can we provide a generic command for that? Maybe something like: 


```suggestion
                remediations="Install the efibootmgr utility using the following command:\n\n 1. yum install efibootmgr",
```